File: member/src/main/resources/static/js/match.js
document.addEventListener('DOMContentLoaded', () => {
    const seasonSelect = document.getElementById('season');
    const teamSelect = document.getElementById('team');
    const matchData = document.getElementById('match-data');

    function updateTeams() {
        const season = seasonSelect.value;
        if (!season) return;

        fetch(`/api/teams?season=${season}`)
            .then(response => response.json())
            .then(teams => {
                teamSelect.innerHTML = '<option value="">-- 선택 --</option>';
                teams.forEach(team => {
                    const option = document.createElement('option');
                    option.value = team;
                    option.textContent = team;
                    teamSelect.appendChild(option);
                });
            });
    }

    function fetchMatches() {
        const season = seasonSelect.value;
        const team = teamSelect.value;

        if (!season || !team) return;

        fetch(`/api/matches?season=${season}&team=${team}`)
            .then(response => response.json())
            .then(data => {
                if (data.length === 0) {
                    matchData.innerHTML = '<tr><td colspan="20">데이터가 없습니다.</td></tr>';
                    return;
                }
                
                matchData.innerHTML = data.map(match => `
                    <tr>
                        <td>${match.season}</td>
                        <td>${match.date}</td>
                        <td>${match.team}</td>
                        <td>${match.teamScore}</td>
                        <td>${match.opponentScore}</td>
                        <td>${match.opponentTeam}</td>
                        <td>${match.champion}</td>
                        <td>${match.opponentChampion}</td>
                        <td>${match.player}</td>
                        <td>${match.opponentPlayer}</td>
                        <td>${match.kda}</td>
                        <td>${match.opponentKda}</td>
                        <td>${match.damage}</td>
                        <td>${match.opponentDamage}</td>
                        <td>${match.sight}</td>
                        <td>${match.opponentSight}</td>
                        <td>${match.cs}</td>
                        <td>${match.opponentCs}</td>
                        <td>${match.perCs}</td>
                        <td>${match.opponentPerCs}</td>
                    </tr>
                `).join('');
            });
    }

    seasonSelect.addEventListener('change', updateTeams);
    teamSelect.addEventListener('change', fetchMatches);
});

File: member/src/main/java/com/example/member/model/Match.java
package com.example.member.model;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name= "2023_LCK_1")
public class Match {
    @Id
    private Long id;
    private String season;
    private String date;
    private String team;
    private Integer teamScore;
    private Integer opponentScore;
    private String opponentTeam;
    private String champion;
    private String opponentChampion;
    private String player;
    private String opponentPlayer;
    private String kda;
    private String opponentKda;
    private String damage;
    private String opponentDamage;
    private String sight;
    private String opponentSight;
    private Integer cs;
    private Integer opponentCs;
    private String perCs;
    private String opponentPerCs;

    // Getters and Setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getSeason() {
        return season;
    }

    public void setSeason(String season) {
        this.season = season;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getTeam() {
        return team;
    }

    public void setTeam(String team) {
        this.team = team;
    }

    public Integer getTeamScore() {
        return teamScore;
    }

    public void setTeamScore(Integer teamScore) {
        this.teamScore = teamScore;
    }

    public Integer getOpponentScore() {
        return opponentScore;
    }

    public void setOpponentScore(Integer opponentScore) {
        this.opponentScore = opponentScore;
    }

    public String getOpponentTeam() {
        return opponentTeam;
    }

    public void setOpponentTeam(String opponentTeam) {
        this.opponentTeam = opponentTeam;
    }

    public String getChampion() {
        return champion;
    }

    public void setChampion(String champion) {
        this.champion = champion;
    }

    public String getOpponentChampion() {
        return opponentChampion;
    }

    public void setOpponentChampion(String opponentChampion) {
        this.opponentChampion = opponentChampion;
    }

    public String getPlayer() {
        return player;
    }

    public void setPlayer(String player) {
        this.player = player;
    }

    public String getOpponentPlayer() {
        return opponentPlayer;
    }

    public void setOpponentPlayer(String opponentPlayer) {
        this.opponentPlayer = opponentPlayer;
    }

    public String getKda() {
        return kda;
    }

    public void setKda(String kda) {
        this.kda = kda;
    }

    public String getOpponentKda() {
        return opponentKda;
    }

    public void setOpponentKda(String opponentKda) {
        this.opponentKda = opponentKda;
    }

    public String getDamage() {
        return damage;
    }

    public void setDamage(String damage) {
        this.damage = damage;
    }

    public String getOpponentDamage() {
        return opponentDamage;
    }

    public void setOpponentDamage(String opponentDamage) {
        this.opponentDamage = opponentDamage;
    }

    public String getSight() {
        return sight;
    }

    public void setSight(String sight) {
        this.sight = sight;
    }

    public String getOpponentSight() {
        return opponentSight;
    }

    public void setOpponentSight(String opponentSight) {
        this.opponentSight = opponentSight;
    }

    public Integer getCs() {
        return cs;
    }

    public void setCs(Integer cs) {
        this.cs = cs;
    }

    public Integer getOpponentCs() {
        return opponentCs;
    }

    public void setOpponentCs(Integer opponentCs) {
        this.opponentCs = opponentCs;
    }

    public String getPerCs() {
        return perCs;
    }

    public void setPerCs(String perCs) {
        this.perCs = perCs;
    }

    public String getOpponentPerCs() {
        return opponentPerCs;
    }

    public void setOpponentPerCs(String opponentPerCs) {
        this.opponentPerCs = opponentPerCs;
    }
}
